src/main/java/org/jenkinsci/plugins/workflow/support/steps/build/RunWrapper.java

```diff
@@ -120,7 +120,17 @@ public void setDisplayName(String n) throws IOException {
             build().setDisplayName(n);
         }
     }
-	
+
+    @Whitelisted
+    public void setProjectDescription(String d) throws IOException {
+        if (!currentBuild) {
```

Comment:
I'm not sure about this. It makes sense that we let a Pipeline change the description/name of its own build, but changing the description of its job, which is shared across all builds, seems different. What is your use case? 

Unfortunately I think more granular access control here would require that something like the authorize-project plugin be installed and configured, so this is basically the best we can do.

Reply:
I understand your position and I could always use JobDSL or similar to do this. Since I already have a workaround using a `@NonCPS` method in my shared library ,I don't really need this to get merged.

Reply:
FWIW I was mostly just curious to understand how you intended to use it; do you have a job that creates other jobs or something?

Reply:
A bit of background: my shared library defines classes for my various build types. I am using Jenkinsfiles as super-config files that create a build object with the given config and then runs it. I've added a config for the project description to some of these Jenkinsfiles so that when the build runs, the project description is updated if the config changes.

As mentioned earlier, I'm currently using a `@NonCPS` method to do this. That method accesses the project instance through `currentBuild.getRawBuild().getParent()` and then updates the description. I wanted to replace that with a single pipeline step and get rid of the `@NonCPS` method.

Reply:
Ok, what you are doing in your shared library seems like a good approach for your use case. @jglick any thoughts on whether exposing the ability to change the description of a job through `RunWrapper` seems ok in general?